lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,137 @@
+from typing import Dict, List, Union
+from abc import ABC, abstractmethod
+
+import os
+from sycamore.data import Element, Document
+from sycamore.evaluation.pipeline import EvaluationPipeline
+from sycamore.evaluation import EvaluationDataPoint
+from sycamore.evaluation.metrics import document_retrieval_metrics, rouge_metrics
+from sycamore import Context
+
+
+def add_searchContext_to_datapoint(datapoint: Document) -> List[Element]:
+    assert isinstance(datapoint, Document)
+    source_documents: List[Element] = []
+    assert datapoint.get("raw") is not None
+    for search_result in datapoint["raw"].get("SearchContexts", []):
```

Comment:
suggestion: replace these with
```python
for search_result in datapoint.get("raw", {}).get("SearchContexts", []):
```
single dict lookup, doesn't crash if "raw" is not present. Although maybe you want that behavior idk?